docs/src/pages/components/data-grid/filtering/QuickFilteringGrid.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import { IconButton, TextField } from '@material-ui/core';
+import {
+  DataGrid,
+  GridToolbarColumnsButton,
+  GridToolbarContainer,
+  GridToolbarDensitySelector,
+  GridToolbarFilterButton,
+} from '@material-ui/data-grid';
+import { useDemoData } from '@material-ui/x-grid-data-generator';
+import ClearIcon from '@material-ui/icons/Clear';
+import SearchIcon from '@material-ui/icons/Search';
+
+const QuickSearchToolbar = (props) => {
+  return (
+    <GridToolbarContainer
+      style={{
+        justifyContent: 'space-between',
+      }}
+    >
+      <div>
+        <GridToolbarColumnsButton />
+        <GridToolbarFilterButton />
+        <GridToolbarDensitySelector />
+      </div>
+      <TextField
+        value={props.value}
+        onChange={props.onChange}
+        placeholder="Search..."
+        InputProps={{
+          startAdornment: <SearchIcon fontSize="small" />,
+          endAdornment: (
+            <IconButton
```

Comment:
AFAIK we also need to label the button to be accessible

```suggestion
            <IconButton
              title="Clear"
              aria-label="Clear"
```